index.js

```diff
@@ -134,11 +134,40 @@ var Router = function(){
 
       var route = Route(path);
       route.fn = fn;
+      route.idx = this.routes.length;
```

Comment:
You'll have to elaborate. This is correct because `route` is about to be pushed onto the `routes` array.